src/puppetlabs/comidi.clj

```diff
@@ -21,17 +21,24 @@
 (def Zipper
   (schema/pred ks/zipper?))
 
-(def http-methods
-  #{:any :get :post :put :delete :head :options})
-
 (def RequestMethod
   (schema/enum :any :get :post :put :delete :head :options))
 
+; Derived from bidi-schema PatternSegment
 (def RegexPatternSegment
-  [(schema/one Pattern "regex") (schema/one schema/Keyword "variable")])
+  (schema/pair schema/Regex "qual" schema/Keyword "id"))
+
+; Derived from bidi-schema by flattening a vector of Patterns
+(def RouteInfoPathSegment
+  (schema/either bidi-schema/Path
+                 bidi-schema/PatternSegment
+                 schema/Bool))
```

Comment:
@scottyw what about the error messages if the schema check fails, though?  I'm definitely not claiming that {{cond}} is more readable / maintainable in the code, just that it generally produces much more useful error messages, and we've mostly agreed that the benefits of that outweigh the suck of the {{cond}} form.